code/projects/app/src/app/shared/image-display/image-display.component.ts

```diff
@@ -1,20 +1,69 @@
-import { Component, effect, ElementRef, input, viewChild } from "@angular/core";
+import {
+  animate,
+  state,
+  style,
+  transition,
+  trigger,
+} from "@angular/animations";
+import {
+  Component,
+  effect,
+  ElementRef,
+  input,
+  signal,
+  viewChild,
+} from "@angular/core";
 import { drawOn } from "../../util/image-data";
+import { ImageComparisonService } from "../image-comparison/image-comparison.service";
+
+type Flash = "red" | "green" | "default";
 
 @Component({
   selector: "app-image-display",
   templateUrl: "./image-display.component.html",
   styleUrl: "./image-display.component.css",
+  animations: [
+    trigger("flash", [
+      state("red", style({ backgroundColor: "var(--app-red)" })),
+      state("green", style({ backgroundColor: "var(--app-green)" })),
+      state("*", style({ backgroundColor: "var(--app-navy)" })),
+      transition("red => *", animate("300ms")),
+      transition("green => *", animate("300ms")),
+      transition("* => red", animate("0ms")),
+      transition("* => green", animate("0ms")),
+    ]),
+  ],
 })
 export class ImageDisplayComponent {
   public readonly image = input<ImageData>();
+  public showButtons = input<boolean>(true);
+  protected showControls = signal(false);
```

Comment:
```suggestion
  public readonly showButtons = input(true);
  protected readonly showControls = signal(false);
```
**nit:** All signals can be readonly since we'd have huge problems if they were ever reassigned (all of their consumers would use the "hanging" reference anyway). Explicit types for `input`s (and generally all signals) are only required if they can't be inferred from the `initialValue`.